File: web/src/api/routing.ts
import apiClient from './client'

export interface RoutingConfig {
  id: number
  judge_model_id: number | null
  default_group_id: number | null
  judge_max_input_chars: number
  gateway_token: string
}

export async function getRoutingConfig(): Promise<RoutingConfig> {
  const { data } = await apiClient.get('/admin/routing')
  return data
}

export async function updateRoutingConfig(rc: Partial<RoutingConfig>): Promise<RoutingConfig> {
  const { data } = await apiClient.put('/admin/routing', rc)
  return data
}

File: internal/store/routing.go
package store

type RoutingConfig struct {
	ID                 uint  `gorm:"primaryKey" json:"id"`
	JudgeGroupID       *uint `json:"judge_group_id"`        // 判定队列，指向 ModelGroup
	DefaultGroupID     *uint `json:"default_group_id"`      // default fallback queue
	JudgeMaxInputChars int   `gorm:"default:2000" json:"judge_max_input_chars"`
}

func (s *Store) GetRoutingConfig() (*RoutingConfig, error) {
	var rc RoutingConfig
	if err := s.DB.First(&rc, 1).Error; err != nil {
		return nil, err
	}
	return &rc, nil
}

// UpdateRoutingConfig saves the routing config singleton. 判定完全由
// judge_group_id 驱动；旧 is_judge 镜像逻辑已随判定队列化移除。
func (s *Store) UpdateRoutingConfig(rc *RoutingConfig) error {
	rc.ID = 1
	return s.DB.Save(rc).Error
}
